libs/core/datastructures/include/hpx/datastructures/detail/small_vector.hpp

```diff
@@ -846,6 +943,10 @@ namespace hpx::detail {
 
         [[nodiscard]] constexpr auto capacity() const noexcept -> std::size_t
         {
+            if constexpr (emulate_inplace_vector)
+            {
+                return capacity<direction::direct>();
+            }
```

Comment:
This seems redundant, since `is_direct()` returns constexpr `true` in that case. Or if you're trying to avoid even instantiating `capacity<direction::indirect>()` in that case, then it ought to be enclosed in an `else` block.